src/rpc/core_rpc_server_commands_defs.h

```diff
@@ -2763,15 +2764,23 @@ namespace cryptonote
         std::string                           operator_address;              // The wallet address of the operator to which the operator cut of the staking reward is sent to.
         std::string                           public_ip;                     // The public ip address of the service node
         uint16_t                              storage_port;                  // The port number associated with the storage server
-        bool                                  storage_server_reachable;      // Whether the node's storage server has been reported as unreachable for a long time
+
+        // Service Node Testing
+        uint64_t                                           last_uptime_proof;                   // The last time this Service Node's uptime proof was relayed by at least 1 Service Node other than itself in unix epoch time.
+        bool                                               storage_server_reachable;            // Whether the node's storage server has been reported as unreachable for a long time
+        uint64_t                                           storage_server_reachable_timestamp;  // The last time this Service Node's storage server was contacted
+        uint16_t                                           version_major;                       // Major version the node is currently running
+        uint16_t                                           version_minor;                       // Minor version the node is currently running
+        uint16_t                                           version_patch;                       // Patch version the node is currently running
+        std::vector<service_nodes::checkpoint_vote_record> votes;                               // Of the last N checkpoints the Service Node is in a checkpointing quorum, record whether or not the Service Node voted to checkpoint a block
```

Comment:
I think we need to deprecate this soon (though not saying in this PR) and stop adding to it and instead move everything to `get_n_service_nodes` so that callers pick-and-choose what they want and so we can stop mostly duplicating these two structures.  But mainly because the default `get_service_nodes` data response is getting *massive*.

Reply:
Agreed. We need to come up with some deprecating protocol for RPC. Various external services are using this endpoint.